migrations/1564410815456-migrateSchoolsTo2019-20.js

```diff
@@ -0,0 +1,43 @@
+const { connect, close } = require('../src/utils/database');
+// We need to import the systems model to register it with mongoose
+/* eslint-disable-next-line no-unused-vars */
+const System = require('../src/services/system/model');
+const { schoolModel: School, yearModel: Year } = require('../src/services/school/model');
+const { schoolUsesLdap } = require('../src/services/school/maintenance');
+const { info } = require('../src/logger');
+
+module.exports = {
+	up: async function up() {
+		await connect();
+
+		info('Fetching next school year...');
+		const nextYear = await Year.findOne({ name: '2019/20' }).lean().exec();
+		info('=>', nextYear);
+
+		info('Fetching schools...');
+		const schools = await School.find({}).select(['name']).populate(['systems']);
+		info(`Got ${schools.length} schools.`);
+
+		for (const school of schools) {
+			info(`Migrating ${school.name} (${school._id})...`);
+			if (schoolUsesLdap(school)) {
+				info('School uses LDAP');
+				// schools with active LDAP systems are set into maintenance mode effective immediately
+				const result = await School.updateOne({ _id: school._id }, { inMaintenanceSince: Date.now() });
```

Comment:
Date.now() schould be one single value by instance. Predefine that value above and log it, by that we can exactly find out, which schools have been updated using our script.

Reply:
Log, if maintainance mode has been overridden.

Reply:
log result.toObject() however the whole mongoose model seems to be huge...

Reply:
`result` is the result of the mongodb write operation, i.e. something like `{"n":1,"nModified":1,"ok":1}`